### PR TITLE
Fixes parallel computation of average formation factors.

### PR DESCRIPTION
**Pull request by blattms**
This was using the sum of the sum of all cells of the partitions. Hence the shared cells were counted twice.

Beware: this might change results (for the better).

### CODE REVIEW

**Comment by blattms** [comment]
jenkins build this please

File: ebos/eclbasevanguard.hh [hunk 1]
```diff
@@ -617,6 +617,31 @@ public:
     {
         return centroids_;
     }
+
+    /*!
+     * \brief Get the number of cells inth global leaf grid view.
```
Comment (akva2) [inline ebos/eclbasevanguard.hh:622]: in the

File: ebos/eclbasevanguard.hh [hunk 2]
```diff
@@ -617,6 +617,31 @@ public:
...
+     * \warn This is a collective operation that needs to be called
+     * on all ranks.
+     */
+    std::size_t noGlobalCells() const
+    {
+        const auto& grid = asImp_().grid();
+        if (grid.comm().size())
```
Comment (bska) [inline ebos/eclbasevanguard.hh:629]: That looks unexpected.  Is it supposed to include `== 0`?  If not, what am I missing?

Comment (akva2) [inline ebos/eclbasevanguard.hh:629]: == 1  ?

Comment (blattms) [reply ebos/eclbasevanguard.hh]: of course

**Comment by blattms** [comment]
I am testing this locally now first.

File: ebos/eclbasevanguard.hh [hunk 3]
```diff
@@ -617,6 +617,31 @@ public:
...
+        const auto& grid = asImp_().grid();
+        if (grid.comm().size() == 1)
+        {
+            return grid.leafGridView().size(0);
+        }
+        const auto& gridView = grid.leafGridView();
+        const auto& elemEndIt = gridView.template end</*codim=*/0, Dune::Interior_Partition>();
+        std::size_t global_nc = 0;
+
+        for (auto elemIt = gridView.template begin</*codim=*/0, Dune::Interior_Partition>();
```
Comment (akva2) [inline ebos/eclbasevanguard.hh:637]: could use std::distance to add some nitpicking.

Comment (blattms) [reply ebos/eclbasevanguard.hh]: Nice idea. Did that.

**Comment by blattms** [comment]
Seems I have removed the obvious bugs.

**Comment by blattms** [comment]
jenkins build this please

**Comment by blattms** [comment]
jenkins build this please

**Comment by blattms** [comment]
Nobody complained about my rather poor naming "noGlobalCells".

**Comment by blattms** [comment]
benchmark please

**Comment by atgeirr** [comment]
> Nobody complained about my rather poor naming: "noGlobalCells".

I'll shoot... When I read the comment here, I though you meant "no", when I looked at the code (which is nice, and well documented BTW, no complaints there) I saw it was "number of". I think "numGlobalCells()" is much better: just one more character, but approximately 93.8923% fewer misunderstandings.

Also, the naming of the variables is unfortunate (not new in this PR). Rather than `number_of_cells_` and `global_nc_` we should have names that make clear their distinctions, what about `local_num_cells_` and `global_num_cells_`? (Just since we are in the area. That could be a separate little PR instead.)

**Comment by ytelses** [comment]
Benchmark result overview: 




| Test    | Configuration                              | Relative |
| ------- | ------------------------------------------ | -------- |
| opm-git | OPM Benchmark: flow_mpi_extra - Threads: 1 | 1        |
| opm-git | OPM Benchmark: flow_mpi_extra - Threads: 8 | 1.021    |
| opm-git | OPM Benchmark: flow_mpi_norne - Threads: 1 | 0.999    |
| opm-git | OPM Benchmark: flow_mpi_norne - Threads: 8 | 0.978    |
* Speed-up = Total time master / Total time pull request. Above 1.0 is an improvement. *



View result details @ https://www.ytelses.com/opm/?page=result&id=915

**Comment by blattms** [comment]
jenkins run norne serial parallel please

**Comment by blattms** [comment]
Mmh, somehow I imagined I would get a comparison of well curves in parallel and serial.

**Comment by akva2** [comment]
jenkins run norne norne_parallel please

**Comment by akva2** [comment]
there u go.

**Comment by atgeirr** [comment]
Looks good, merging.

**Pull request merged**